libraries/engage/filter/providers/FilterPageProvider.jsx

```diff
@@ -0,0 +1,320 @@
+import React, {
+  useMemo, useState, useEffect, useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import debounce from 'lodash/debounce';
+import isEqual from 'lodash/isEqual';
+import { router } from '@shopgate/engage/core';
+import {
+  getFiltersByHash,
+} from '@shopgate/engage/filter';
+// eslint-disable-next-line no-unused-vars, import/named
+import Context, { APIFilter, RouteFilters } from './FilterPageProvider.context';
```

Comment:
APIFilter and RouteFilters are not exported in the context file, how can they be imported here?

Reply:
they are kind of [see here](https://github.com/shopgate/pwa/blob/f4dd631fe042443205ff8a715d435005504a8f74/libraries/engage/filter/providers/FilterPageProvider.context.js#L22). that's one of the tricks to use JSDoc typedefs to get TS type hints in plain js. unfortunately eslint doesn't recognize that.